src/ScrollBox/index.jsx

```diff
@@ -226,284 +106,384 @@ export default class ScrollBox extends React.Component
-            this.innerRef.scrollLeft = scrollLeft + increment;
+            innerRef.current.scrollLeft = dimensions.scrollLeft + increment;
         }
-    }
+    }, [ onClickScrollUp, onClickScrollDown, onClickScrollLeft,
+        onClickScrollRight, dimensions.clientHeight, dimensions.scrollTop,
+        dimensions.clientWidth, dimensions.scrollLeft, scrollAmount ] );
 
-    handleClickTrackX( pos )
+    const handleClickTrackX = useCallback( ( pos ) =>
     {
-        const { scrollAmount } = this.props;
-        const { clientWidth, scrollLeft } = this.state;
-
-        let amount = clientWidth;
+        let amount = dimensions.clientWidth;
         if ( scrollAmount )
         {
             amount = Array.isArray( scrollAmount ) ?
                 scrollAmount[ 0 ] : scrollAmount;
         }
 
-        const increment = pos >= scrollLeft ? amount : -amount;
-        this.innerRef.scrollLeft = scrollLeft + increment;
-    }
+        const increment = pos >= dimensions.scrollLeft ? amount : -amount;
+        innerRef.current.scrollLeft = dimensions.scrollLeft + increment;
+    }, [ dimensions.clientWidth, dimensions.scrollLeft, scrollAmount ] );
 
-    handleClickTrackY( pos )
+    const handleClickTrackY = useCallback( ( pos ) =>
     {
-        const { scrollAmount } = this.props;
-        const { clientHeight, scrollTop } = this.state;
-
-        let amount = clientHeight;
+        let amount = dimensions.clientHeight;
         if ( scrollAmount )
         {
             amount = Array.isArray( scrollAmount ) ?
                 scrollAmount[ 1 ] : scrollAmount;
         }
 
-        const increment = pos >= scrollTop ? amount : -amount;
-        this.innerRef.scrollTop = scrollTop + increment;
-    }
+        const increment = pos >= dimensions.scrollTop ? amount : -amount;
+        innerRef.current.scrollTop = dimensions.scrollTop + increment;
+    }, [ dimensions.clientHeight, dimensions.scrollTop, scrollAmount ] );
 
-    handleChangeX( pos )
+    const handleChangeX = useCallback( ( pos ) =>
     {
-        this.innerRef.scrollLeft = pos;
-    }
+        innerRef.current.scrollLeft = pos;
+    }, [ innerRef.current ] );
 
-    handleChangeY( pos )
+    const handleChangeY = useCallback( ( pos ) =>
     {
-        this.innerRef.scrollTop = pos;
-    }
+        innerRef.current.scrollTop = pos;
+    }, [ innerRef.current ] );
 
-    handleRef( ref )
+    const handleRef = useCallback( ( ref ) =>
     {
-        if ( ref )
+        if ( typeof scrollBoxRef === 'function' )
         {
-            if ( this.props.scrollBoxRef )
-            {
-                this.props.scrollBoxRef.current = ref;
-            }
-
-            this.innerRef = ref;
+            scrollBoxRef( ref );
         }
-    }
 
-    handleScroll()
-    {
-        this.forceUpdate();
-    }
+        innerRef.current = ref;
+    }, [ scrollBoxRef ] );
 
-    canScroll( dir )
+    const handleRenderScrollButton = useCallback( ( dir ) =>
```

Comment:
_If_ we are keeping this function it should be renamed (remove the `handle...` bit, since it is not an event handler).